### PR TITLE
function-reference/query-functions: add has_version flag explanation, bug #307065

### PR DESCRIPTION
**Pull request by wimmuskee**
Gentoo-Bug: https://bugs.gentoo.org/show_bug.cgi?id=307065

Added functionality as described in the bugreport, as well as adding a full example.

### CODE REVIEW

File: function-reference/query-functions/text.xml [hunk 1]
```diff
@@ -102,10 +102,12 @@ query variables and similar state.
   </tr>
   <tr>
     <ti>
-      <c>has_version pkg</c>
+      <c>has_version pkg[flag]</c>
     </ti>
     <ti>
-      True if <c>pkg</c> (can include version specifiers) is installed.
+      True if <c>pkg</c> (can include version specifiers) is installed. The use of <c>flag</c> is
+      optional and <c>flag</c> can be a list of multiple USE flags. Example:
```
Comment (gktrk) [inline function-reference/query-functions/text.xml]: `list of multiple USE flags` can be simplified to `list of USE flags`

File: function-reference/query-functions/text.xml [hunk 2]
```diff
@@ -102,10 +102,12 @@ query variables and similar state.
...
+      <c>if has_version "=app-editors/nano-2.5.3"["nls,spell"] ; then</c>.
```
Comment (gktrk) [inline function-reference/query-functions/text.xml]: the entire `pkg[flag]` can be contained in `""`

File: function-reference/query-functions/text.xml (hunk 1 above)
Comment (gktrk) [inline function-reference/query-functions/text.xml]: hmm, now that I think about it again, the proper name for those is [built with use dependencies](https://devmanual.gentoo.org/general-concepts/dependencies/index.html#built-with-use-dependencies). Maybe we can say something like: `True if <c>pkg</c> (can include version specifiers and built with use dependencies) is installed. Example:` ....

File: function-reference/query-functions/text.xml (hunk 2 above)
Comment (wimmuskee) [inline function-reference/query-functions/text.xml]: would you like me to do an uri refer to the "built with use dependencies" page?

**Comment by wimmuskee** [comment]
would you like me to do an uri refer to the "built with use dependencies" page?

**Comment by wimmuskee** [comment]
updated with regard to your comments, feel free to squash to one commit
and thx for the comments

**Comment by gktrk** [comment]
Merged in https://github.com/gentoo/devmanual.gentoo.org/commit/9101c681314f17ff94abc075afc1d80c42742383. Thanks!

**Pull request closed without merging**